### PR TITLE
Do batch updates of WorkCoverageRecords in ascending order

### PR DESCRIPTION
**Pull request by vbessonov**
## Description

This PR forces CM to do bulk updates of `WorkCoverageRecords` in ascending order. 
_I used recommendations from this [SO article](https://stackoverflow.com/questions/44660368/postgres-update-with-order-by-how-to-do-it)._

I fixed only `WorkCoverageRecord.bulk_add`, but there is also `CoverageRecord.bulk_add`. I haven't touched it yet because I didn't see issues related with `CoverageRecord` we may need to change it as well.

## How Has This Been Tested?

I ran `work_classification` and `search_index_refresh` simultaneously locally.

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: model/coverage.py [hunk 1]
```diff
@@ -664,9 +663,15 @@ def bulk_add(
...
-                    WorkCoverageRecord.operation == operation,
+                WorkCoverageRecord.id.in_(
+                    _db.query(WorkCoverageRecord.id)
+                    .with_for_update()
+                    .filter(
+                        and_(
+                            WorkCoverageRecord.work_id.in_(work_ids),
+                            WorkCoverageRecord.operation == operation,
+                        )
+                    )
```
Comment (tdilauro) [inline model/coverage.py:674]: Do you need an `order_by(…)` in here, as well?

File: model/coverage.py [hunk 2]
```diff
@@ -664,9 +663,15 @@ def bulk_add(
         update = (
             WorkCoverageRecord.__table__.update()
             .where(
-                and_(
-                    WorkCoverageRecord.work_id.in_(work_ids),
-                    WorkCoverageRecord.operation == operation,
+                WorkCoverageRecord.id.in_(
```
Comment (tdilauro) [inline model/coverage.py:666]: It might be helpful to add a comment about what you are aiming to do with this more “involved” update.

**Review by tdilauro — COMMENTED** [review]
In the main comment, you mentioned that you were able to run `work_classification` and `search_index_refresh` simultaneously. Were you not able to run them simultaneously before the change?

File: model/coverage.py (hunk 1 above)
Comment (vbessonov) [reply model/coverage.py:674]: `work_ids` contains already sorted array. The list of `Work` objects was sorted before in `WorkCoverageProvider.items_that_need_coverage` inside `Work.missing_coverage_from`.

File: model/coverage.py (hunk 2 above)
Comment (vbessonov) [reply model/coverage.py:670]: I added a comment.

**Review by tdilauro — APPROVED** [review]
LGTM! ⭐

**Pull request merged**